### PR TITLE
Adds ability to auto approve and merge Dependabot PRs

### PR DESCRIPTION
**Pull request by allez-allez-allez**

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      review:
+        type: APPROVE
+
+  - name: write issue comment to start codebuild job
+    conditions:
+      - author~=dependabot\[bot\]|dependabot-preview\[bot\]
+      - "#approved-reviews-by>=1"
+    actions:
+      comment:
+        message: go codebuild go
```
Comment (lorengordon) [inline .mergify.yml]: Can we gate this to when a salt-formula is what is getting updated?

Comment (allez-allez-allez) [reply .mergify.yml]: are you referring to these? https://github.com/plus3it/watchmaker/tree/develop/src/watchmaker/static/salt/formulas

trying to look for an example of dependabot pr that updates a salt-forumla

Comment (lorengordon) [reply .mergify.yml]: https://github.com/plus3it/watchmaker/pull/986

**Comment by allez-allez-allez** [comment]
> Just to clarify, this will automatically approve dependabot PRs but they must pass codebuild? It looks like it will automatically make `go codebuild go` comment? Is anyone updating the hook to allow dependabot build privileges?

It will make `go codebuild go` comment only if the pr has passed all the `continuous-integration/*` and `codecov/*` checks.

also, we also need to update the gihub-webhook to allow `mergify[bot]` to trigger codebuild jobs.

File: .mergify.yml (hunk 1 above)
Comment (allez-allez-allez) [reply .mergify.yml]: Proposing the following flow:
* For salt-formula updates PR:
  1) Match title of the title, if begin with `Bump src/watchmaker/static/salt/formulas/`, then create a label called `salt-formula-update`
  2) Create a comment of `go codebuil go` if pr has `salt-formula-update` label 
  3) Auto merge the pr if the codebuild job ran successfully, and the pr has `salt-formula-update` label

* For other dependabots pr, auto-merge after all other CI checks passed.

Comment (lorengordon) [reply .mergify.yml]: Oh, good idea, yeah, let's use the label applied by dependabot.

**Comment by allez-allez-allez** [comment]
Gated the `go codebuild go` comment action for PRs with `submodules` label.

**Comment by YakDriver** [comment]
I'll update the webhook

**Pull request merged**